src/graph/graphcontext.c

```diff
@@ -557,6 +557,17 @@ static void _GraphContext_Free(void *arg) {
 	if(GraphDecodeContext_Finished(gc->decoding_context)) Graph_Free(gc->g);
 	else Graph_PartialFree(gc->g);
 
+
+	bool async_delete;
+	Config_Option_get(Config_ASYNC_DELETE, &async_delete);
+	
+	RedisModuleCtx *ctx = NULL;
+	int success = REDISMODULE_ERR;
+	if(async_delete) {
```

Comment:
I'm missing documentation explaining why we need to acquire GIL at this point

Reply:
@swilly22 `RediSearch_DropIndex` is modifying several global data structures and calling on a thread w/o acquiring GIL might corrupt those structures. The failure that we see repetitively is when adding or removing callbacks from Redis's `Timer`.

Note that `DropIndex` uses a thread internally to free unlinked data w/o holding GIL.

Reply:
added